WooCommerce/Classes/ViewRelated/Blaze/AdDestination/BlazeAdDestinationSettingViewModel.swift

```diff
@@ -0,0 +1,113 @@
+import Foundation
+
+final class BlazeAdDestinationSettingViewModel: ObservableObject {
+    enum DestinationURLType {
+        case product
+        case home
+    }
+
+    var productURL: String
+    var homeURL: String
+
+    @Published var selectedDestinationType: DestinationURLType
+
+    @Published var parameters: [BlazeAdURLParameter]
+
+    // Text to be shown on the view for remaining available characters for custom added parameters.
+    var remainingCharactersLabel: String {
+        let remainingCharacters = calculateRemainingCharacters()
+        let lengthText = String.pluralize(remainingCharacters,
+                                             singular: Localization.parameterCharactersLimit.singular,
+                                             plural: Localization.parameterCharactersLimit.plural)
+        return String(format: lengthText, remainingCharacters)
+    }
+
+    // Text to be shown on the view for the final ad campaign URL including parameters, if any.
+    var finalDestinationLabel: String {
+        return String(format: Localization.finalDestination, buildFinalDestinationURL())
+    }
+
+    init (productURL: String,
+          homeURL: String,
+          selectedDestinationType: DestinationURLType = .product,
+          parameters: [BlazeAdURLParameter] = [BlazeAdURLParameter(key: "key1", value: "value1")]) {
+        self.productURL = productURL
+        self.homeURL = homeURL
+        self.selectedDestinationType = selectedDestinationType
+        self.parameters = parameters
+    }
+
+    func setDestinationType(type: DestinationURLType) {
+        selectedDestinationType = type
+    }
+
+    private func buildParameterString() -> String {
+        var parameterString = ""
+        for parameter in parameters {
+            // In URL format, the parameter is written such as "key=value".
+            parameterString += parameter.key + "=" + parameter.value
+
+            // If it's not the last parameter, add an ampersand.
+            if parameter != parameters.last {
+                parameterString += "&"
+            }
+        }
+
+        return parameterString
+    }
+
+    private func buildFinalDestinationURL() -> String {
+        let baseURL: String
+        switch selectedDestinationType {
+        case .product:
+            baseURL = productURL
+        case .home:
+            baseURL = homeURL
+        }
+
+        let paramString = buildParameterString()
+        return baseURL + (paramString.isEmpty ? "" : "?\(paramString)")
+    }
+
+    private func calculateRemainingCharacters() -> Int {
+        let remainingCharacters = Constant.maxParameterLength - buildParameterString().count
+        // Should stop at zero and not show negative number.
+        return max(0, remainingCharacters)
+    }
+
+    struct BlazeAdURLParameter: Equatable, Hashable {
+        var key: String
+        var value: String
```

Comment:
❓ Any reason for these to be variables and not constants?

Reply:
Thanks, was a wrong copy-paste. https://github.com/woocommerce/woocommerce-ios/pull/11634/commits/db6d891fabfed277442e3e7c28f6fa25881d1519